src/cache_entry.cc

```diff
@@ -308,37 +325,30 @@ CacheEntryItem::FromBytes(boost::span<const std::byte> packed_bytes)
   memcpy(&output_byte_size, packed_bytes.begin() + position, sizeof(uint64_t));
   position += sizeof(uint64_t);
 
-  Buffer output_buffer(output_byte_size);
-  memcpy(
-      output_buffer.data(), packed_bytes.begin() + position, output_byte_size);
+  // NOTE: Reference buffer section of packed bytes directly, DO NOT copy here.
+  // We will copy this buffer into the response object in ToResponse, so the
+  // buffer must remain valid until then. They should remain valid until the
+  // CacheEntryItem is destructed.
+  const void* output_buffer =
+      static_cast<const void*>(packed_bytes.data() + position);
   position += output_byte_size;
 
+  // Verify packed bytes matched expected size before allocating output_buffer
+  auto output = CacheOutput();
   if (packed_bytes.begin() + position != packed_bytes.end()) {
-    LOG_ERROR << "Unexpected number of bytes. Received " << packed_bytes.size()
-              << ", expected: " << position;
-    return std::nullopt;
-  }
-
-  // TODO: Remove
-  LOG_VERBOSE(2) << "[DECODE] name_byte_size=" << name_byte_size;
-  LOG_VERBOSE(2) << "[DECODE] name=" << name;
-  LOG_VERBOSE(2) << "[DECODE] dtype_byte_size=" << dtype_byte_size;
-  LOG_VERBOSE(2) << "[DECODE] dtype=" << dtype;
-  LOG_VERBOSE(2) << "[DECODE] shape_byte_size=" << shape_byte_size;
-  LOG_VERBOSE(2) << "[DECODE] shape=";
-  for (const auto& dim : shape) {
-    LOG_VERBOSE(2) << dim << " ";
+    auto fail = Status(
+        Status::Code::INTERNAL, "Unexpected number of bytes received: " +
+                                    std::to_string(packed_bytes.size()) +
+                                    ", expected: " + std::to_string(position));
+    return {fail, output};
```

Comment:
nit: I think it would be better to just return the error instead of pair and get the output as one of the arguments. 1. the memory copies are more clear. 2. We can use the `RETURN_IF_ERROR` macros. It can be fixed in a future PR.

Reply:
Sure, will do